classes/class-wc-connect-paypal-ec.php

```diff
@@ -0,0 +1,320 @@
+					add_filter( 'woocommerce_payment_gateway_supports', array( $this, 'ppec_supports' ), 10, 3 );
+					add_filter( 'wc_services_pointer_post.php', array( $this, 'register_refund_pointer' ) );
+
+					if ( 'live' === $settings->environment ) {
+						add_action( 'woocommerce_order_status_on-hold', array( $this, 'maybe_set_banner' ) );
+						add_action( 'woocommerce_payment_complete', array( $this, 'maybe_set_banner' ) );
+						add_action( 'admin_enqueue_scripts', array( $this, 'maybe_init_banner' ) );
+					}
+					add_filter( 'pre_option_wc_gateway_ppce_prompt_to_connect', '__return_empty_string' ); // Disable default PPEC notice.
+				}
+			}
+		}
+
+		/**
+		 * Attach request proxying hook if it's an Express Checkout method
+		 */
+		public function request_body( $body ) {
+			if ( in_array( $body['METHOD'], $this->methods_to_proxy ) ) {
+				add_filter( 'pre_http_request', array( $this, 'proxy_request' ), 10, 3 );
+			} else {
+				remove_filter( 'pre_http_request', array( $this, 'proxy_request' ), 10, 3 );
+			}
+			return $body;
+		}
+
+		/**
+		 * Reroute Express Checkout requests from the PPEC extension via WCS server to pick up API credentials
+		 */
+		public function proxy_request( $preempt, $r, $url ) {
+			if ( ! preg_match( '/paypal.com\/nvp$/', $url ) ) {
+				return $preempt;
+			}
+
+			$settings = wc_gateway_ppec()->settings;
+			return $this->api_client->proxy_request( 'paypal/nvp/' . $settings->environment, $r );
+		}
+
+		/**
+		 * Limit supported payment gateway features to payments
+		 */
+		public function ppec_supports( $supported, $feature, $gateway ) {
+			return 'ppec_paypal' === $gateway->id ? 'products' === $feature : $supported;
+		}
+
+		/**
+		 * Add a pointer clarifying the need to link an account before refunding payment
+		 */
+		public function register_refund_pointer( $pointers ) {
+			$pointers[] = array(
+				'id' => 'wc_services_refund_via_ppec',
+				'target' => '.refund-actions > button:first-child',
+				'options' => array(
+					'content' => sprintf( '<h3>%s</h3><p>%s</p>',
+						__( 'Link a PayPal account' ,'woocommerce-services' ),
+						sprintf(
+							wp_kses(
+								__( 'To issue refunds via PayPal Express Checkout, you will need to <a href="%s">link a PayPal account</a> with the email address that received this payment.', 'woocommerce-services' ),
+								array(  'a' => array( 'href' => array() ) )
+							),
+							wc_gateway_ppec()->ips->get_signup_url( wc_gateway_ppec()->settings->environment )
+						)
+					),
+					'position' => array( 'edge' => 'bottom', 'align' => 'top' ),
+				),
+				'delayed_opening' => array(
+					'show_button' => '.refund-items',
+					'hide_button' => '.cancel-action',
+					'animating_container' => '.wc-order-refund-items',
+					'delegation_container' => '#woocommerce-order-items',
+				),
+			);
+
+			return $pointers;
+		}
+
+		/**
+		 * If PPEC order comes in, activate prompt to connect a PayPal account
```

Comment:
Consider moving this comment to where the action is attached, and rewriting this to be more specific to the body of the function here.

_E.g. this function isn't tied to an order coming in_